4D-API_cs2cs-style.gie fails on Debian unstable (with GCC 8)

`4D-API_cs2cs-style.gie` started to fail on Debian unstable recently. Quite possibly related to GCC 8 becoming the default compiler.

```
-------------------------------------------------------------------------------
Reading file 'GDA.gie'
-------------------------------------------------------------------------------
total:  3 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
PROJ_LIB=../../nad ../../src/gie builtins.gie
-------------------------------------------------------------------------------
Reading file 'axisswap.gie'
-------------------------------------------------------------------------------
total: 27 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
PROJ_LIB=../../nad ../../src/gie deformation.gie
-------------------------------------------------------------------------------
Reading file '4D-API_cs2cs-style.gie'
     -----
     FAILURE in 4D-API_cs2cs-style.gie(114):
     expected: 12.5 55.5 0
     got:      inf   inf   inf
     deviation:  999999999.999000 mm,  expected:  150.000000 mm
-------------------------------------------------------------------------------
total: 28 tests succeeded,  0 tests skipped,  1 tests FAILED!
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
Reading file 'deformation.gie'
-------------------------------------------------------------------------------
total: 37 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
make[5]: *** [Makefile:455: 4D-API-cs2cs-style] Error 1
make[5]: *** Waiting for unfinished jobs....
-------------------------------------------------------------------------------
Reading file 'builtins.gie'
-------------------------------------------------------------------------------
total: 1568 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
```

--- a/src/pj_param.c
+++ b/src/pj_param.c
@@ -42,7 +42,7 @@ paralist *pj_mkparam_ws (char *str) {
     }
 
     /* Use calloc to automagically 0-terminate the copy */
-    newitem = (paralist *) pj_calloc (1, sizeof(paralist) + len);
+    newitem = (paralist *) pj_calloc (1, sizeof(paralist) + len + 1);
     if (0==newitem)
         return 0;
     memmove(newitem->param, str, len);

--- a/src/projects.h
+++ b/src/projects.h
@@ -458,7 +458,12 @@ struct PJconsts {
 struct ARG_list {
     paralist *next;
     char used;
-    char param[1];
+#ifdef __GNUC__
+    char param[0]; /* variable-length member */
+    /* Safer to use [0] for gcc. See https://github.com/OSGeo/proj.4/pull/1087 */
+#else
+    char param[1]; /* variable-length member */
+#endif
 };
 
 

